kratos/sources/model_part_io.cpp

```diff
@@ -288,7 +288,7 @@ void ModelPartIO::WriteGeometries(GeometryContainerType const& rThisGeometries)
         (*mpStream) << "Begin Geometries\t" << geometry_name << std::endl;
         const auto it_geom_begin = rThisGeometries.Geometries().begin();
         (*mpStream) << "\t" << it_geom_begin->Id() << "\t";
-        auto& r_geometry = *(it_geom_begin.base()->second);
+        auto& r_geometry = *(*(it_geom_begin.base()));
```

Comment:
shouldn't be: `auto& r_geometry = *it_geom_begin`

Reply:
I can give it a try, but I think that this will give you the content of the `begin()` iterator of the underlying container, which is an `std::vector` or, in other words, a pointer to the geometry but not a reference to the geometry.

Reply:
Please give it a try! The PointerVectorSet has its * operator redefined.